pkg/deployment/ipam.go

```diff
@@ -31,74 +32,64 @@ import (
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
 )
 
-// EnsureIPSets Creates the IPSets
-func EnsureIPSets(ctx context.Context, helper *helper.Helper,
-	instance *dataplanev1.OpenStackDataPlaneNodeSet,
-) (map[string]infranetworkv1.IPSet, bool, error) {
-	allIPSets, err := reserveIPs(ctx, helper, instance)
-	if err != nil {
-		instance.Status.Conditions.MarkFalse(
-			dataplanev1.NodeSetIPReservationReadyCondition,
-			condition.ErrorReason, condition.SeverityError,
-			dataplanev1.NodeSetIPReservationReadyErrorMessage)
-		return nil, false, err
-	}
-
-	if len(allIPSets) == 0 {
-		return nil, true, nil
-	}
-
-	for _, s := range allIPSets {
-		if s.Status.Conditions.IsFalse(condition.ReadyCondition) {
-			instance.Status.Conditions.MarkFalse(
-				dataplanev1.NodeSetIPReservationReadyCondition,
-				condition.RequestedReason, condition.SeverityInfo,
-				dataplanev1.NodeSetIPReservationReadyWaitingMessage)
-			return nil, false, nil
-		}
-	}
-	instance.Status.Conditions.MarkTrue(
-		dataplanev1.NodeSetIPReservationReadyCondition,
-		dataplanev1.NodeSetIPReservationReadyMessage)
-	return allIPSets, true, nil
-}
-
-// DataplaneDNSData holds information relevant to the OpenStack DNS configuration of the cluster.
-type DataplaneDNSData struct {
```

Comment:
Is changing the struct name and order of fields really necessary? The patch could be substantially smaller/cleaner with just the field addition. Furthermore, I would like to see the comment amended, not removed. Code should be made more readable, not less.

Reply:
DNSData means[1] something else and it's confusing to use DataplaneDnsData along with it in the same code. The comment is partially incorrect (does not have DNS configuration, it's just a struct used internally for convience) and not an exposed type.

As per why not using reciever, it only makes sense to use a receiver if the function is part of an interface implementation.

[1] https://github.com/openstack-k8s-operators/infra-operator/blob/main/apis/network/v1beta1/dnsdata_types.go#L68

Reply:
https://grisha.org/blog/2016/09/22/golang-receiver-vs-function/

Reply:
Where are we going to stub this function? Or are we going to do that in the future? Other than that the article doesn't have much in a way of argument. It even admits the result looks worse:

> the only benefit you are getting from using a method is syntactic sweetness.

Reply:
It provides only one argument, not valid in our case, as to why it may be better to use functions. And then it says the result looks worse if you do. In my book that's not exactly an endorsement of the "function first" approach. 

Also, I do consider populating struct fields to be a type of operation. But that may just be me.